docs/source/conf.py

```diff
@@ -101,6 +101,41 @@
     new_main_rst.write("".join(main_rst_content[2:]))
     new_main_rst.close()
 
+# Generate optional-plugins
+optional_plugins_path = os.path.join(root_path, "optional_plugins")
+api_optional_plugins_path = os.path.join(base_api_output_dir,
+                                         "optional-plugins")
+if not os.path.exists(api_optional_plugins_path):
+    os.makedirs(api_optional_plugins_path)
+with open(os.path.join(api_optional_plugins_path, "index.rst"),
+          'w') as optional_plugins_toc:
+    optional_plugins_toc.write(""".. index file for optional plugins API
+
+====================
+Optional Plugins API
+====================
+
+The following pages are auto-generated API documentation of optional
+Avocado plugins. This is not public API, it represents only current
+version and can be changed any time.
+
+.. toctree::
+   :maxdepth: 1
+
+    """)
+    for path in os.walk(optional_plugins_path).next()[1]:
+        name = "avocado_%s" % os.path.basename(path)
+        path = os.path.join(optional_plugins_path, path, name)
+        if not os.path.exists(path):
+            continue
+        output_dir = os.path.join(api_optional_plugins_path, name)
+        params = {"api_source_dir": path, "output_dir": output_dir,
+                  "exclude_dirs": ""}
+        process.run(apidoc_template % params)
+        # Remove the unnecessary generated files
+        os.unlink(os.path.join(output_dir, "modules.rst"))
+        optional_plugins_toc.write("\n   %s" % os.path.join(name, name))
```

Comment:
well then we need to run in on RTD. As for the local build I think it's expected for users to run `make develop` manually (or we can add basic `make html` target to `Makefile` for convenience, but I doubt many people will have problem with this...